# Importation de librairies et lectures de fichiers 

In [183]:
import pandas as pd
import numpy as np
import re
import matplotlib.pyplot as plt
import seaborn as sns

In [185]:
# lecture fichiers
web=pd.read_excel('C:\\Users\\agued\\OneDrive\\Escritorio\\web.xlsx')
liaison=pd.read_excel ('C:\\Users\\agued\\OneDrive\\Escritorio\\liaison.xlsx')
erp=pd.read_excel('C:\\Users\\agued\\OneDrive\\Escritorio\\erp.xlsx')

# Preparation de données
## web

In [29]:
web.head()

sku  virtual  downloadable  rating_count  average_rating  \
0  bon-cadeau-25-euros        0             0             0             0.0   
1                15298        0             0             0             0.0   
2                15296        0             0             0             0.0   
3                15300        0             0             0             0.0   
4                19814        0             0             0             0.0   

   total_sales tax_status  tax_class  post_author           post_date  ...  \
0         10.0    taxable        NaN          1.0 2018-06-01 13:53:46  ...   
1          6.0    taxable        NaN          2.0 2018-02-08 12:58:52  ...   
2          0.0    taxable        NaN          2.0 2018-02-08 13:49:41  ...   
3          0.0    taxable        NaN          2.0 2018-02-08 14:08:36  ...   
4          3.0    taxable        NaN          2.0 2018-02-09 14:01:05  ...   

                                           post_name       post_modified  \
0                             bon-cadeau-de-25-euros 2018-06-01 14:13:57   
1        pierre-jean-villa-saint-joseph-preface-2018 2019-12-30 09:30:29   
2          pierre-jean-villa-saint-joseph-tilde-2017 2019-12-21 09:00:17   
3  pierre-jean-villa-croze-hermitage-accroche-coe... 2020-06-26 18:15:03   
4                  pierre-jean-villa-igp-gamine-2018 2020-01-04 16:36:01   

    post_modified_gmt post_content_filtered post_parent  \
0 2018-06-01 12:13:57                   NaN         0.0   
1 2019-12-30 08:30:29                   NaN         0.0   
2 2019-12-21 08:00:17                   NaN         0.0   
3 2020-06-26 16:15:03                   NaN         0.0   
4 2020-01-04 15:36:01                   NaN         0.0   

                                                guid menu_order  post_type  \
0  https://www.bottle-neck.fr/?post_type=product&...        0.0    product   
1  https://www.bottle-neck.fr/?post_type=product&...        0.0    product   
2  https://www.bottle-neck.fr/?post_type=product&...        0.0    product   
3  https://www.bottle-neck.fr/?post_type=product&...        0.0    product   
4  https://www.bottle-neck.fr/?post_type=product&...        0.0    product   

  post_mime_type comment_count  
0            NaN           0.0  
1            NaN           0.0  
2            NaN           0.0  
3            NaN           0.0  
4            NaN           0.0  

[5 rows x 28 columns]

In [30]:
web.shape

(1513, 28)

In [38]:
web.isnull().sum()


sku                        85
virtual                     0
downloadable                0
rating_count                0
average_rating             83
total_sales                83
tax_status                797
tax_class                1513
post_author                83
post_date                  83
post_date_gmt              83
post_content             1513
post_title                 83
post_excerpt              797
post_status                83
comment_status             83
ping_status                83
post_password            1513
post_name                  83
post_modified              83
post_modified_gmt          83
post_content_filtered    1513
post_parent                83
guid                       83
menu_order                 83
post_type                  83
post_mime_type            799
comment_count              83
dtype: int64

In [39]:
web.nunique().tolist

<bound method IndexOpsMixin.tolist of sku                       714
virtual                     1
downloadable                1
rating_count                1
average_rating              1
total_sales                41
tax_status                  1
tax_class                   0
post_author                 2
post_date                 716
post_date_gmt             716
post_content                0
post_title                713
post_excerpt              679
post_status                 1
comment_status              1
ping_status                 1
post_password               0
post_name                 716
post_modified             589
post_modified_gmt         589
post_content_filtered       0
post_parent                 1
guid                     1430
menu_order                  1
post_type                   2
post_mime_type              1
comment_count               1
dtype: int64>

In [186]:
web.drop(['tax_class', 'post_content','post_password','post_content_filtered','virtual','downloadable','rating_count'], axis = 1, inplace = True)

### *on a vu que la colonne 'guid' peut servir de clé aussi avec 1430 valeurs uniques, donc je constate combien de lignes il y a avec un valeur pour 'guid' avec sku null

In [187]:
web[(web['guid'].notnull())& (web['sku'].isnull())]

sku  average_rating  total_sales tax_status  post_author  \
470  NaN             0.0          0.0    taxable          2.0   
471  NaN             0.0          0.0    taxable          2.0   

              post_date       post_date_gmt  \
470 2018-07-31 12:07:23 2018-07-31 10:07:23   
471 2018-08-08 11:23:43 2018-08-08 09:23:43   

                                          post_title  \
470       Pierre Jean Villa Côte Rôtie Fongeant 2017   
471  Pierre Jean Villa Condrieu Jardin Suspendu 2018   

                                          post_excerpt post_status  ...  \
470  Fongeant 2017 explose sur un fruit brillant, p...     publish  ...   
471  <span id="u1194-83">Le nez séduit par ses parf...     publish  ...   

    ping_status                                   post_name  \
470      closed  pierre-jean-villa-cote-rotie-fongeant-2017   
471      closed    pierre-jean-villa-condrieu-suspendu-2018   

          post_modified   post_modified_gmt post_parent  \
470 2019-11-02 13:24:15 2019-11-02 12:24:15         0.0   
471 2019-11-02 13:24:01 2019-11-02 12:24:01         0.0   

                                                  guid menu_order  post_type  \
470  https://www.bottle-neck.fr/?post_type=product&...        0.0    product   
471  https://www.bottle-neck.fr/?post_type=product&...        0.0    product   

    post_mime_type comment_count  
470            NaN           0.0  
471            NaN           0.0  

[2 rows x 21 columns]

il y a deux lignes mais sans aucune ventes (total_sales), 

la colonne Sku sera notre cle pour les jointures, donc je supprime les valeurs nulls, et verifie les doublons

In [188]:
web=web.dropna(subset=['sku'])

In [189]:
web.duplicated('sku').sum() 

714

Analyse de differences entre les doublons. 

In [73]:
values=[15296,19814]
web[web.sku.isin(values)]

sku  average_rating  total_sales tax_status  post_author  \
2    15296             0.0          0.0    taxable          2.0   
4    19814             0.0          3.0    taxable          2.0   
800  15296             0.0          0.0        NaN          2.0   
802  19814             0.0          3.0        NaN          2.0   

              post_date       post_date_gmt  \
2   2018-02-08 13:49:41 2018-02-08 12:49:41   
4   2018-02-09 14:01:05 2018-02-09 13:01:05   
800 2018-02-08 13:49:41 2018-02-08 12:49:41   
802 2018-02-09 14:01:05 2018-02-09 13:01:05   

                                            post_title  \
2      Pierre Jean Villa Saint-Joseph Rouge Tildé 2017   
4    Pierre Jean Villa IGP Collines Rhodaniennes Ga...   
800    Pierre Jean Villa Saint-Joseph Rouge Tildé 2017   
802  Pierre Jean Villa IGP Collines Rhodaniennes Ga...   

                                          post_excerpt post_status  ...  \
2    Les vieilles vignes lui apportent une rare pro...     publish  ...   
4    Gamine représente tout le fruité et la gourman...     publish  ...   
800                                                NaN     publish  ...   
802                                                NaN     publish  ...   

    ping_status                                  post_name  \
2        closed  pierre-jean-villa-saint-joseph-tilde-2017   
4        closed          pierre-jean-villa-igp-gamine-2018   
800      closed  pierre-jean-villa-saint-joseph-tilde-2017   
802      closed          pierre-jean-villa-igp-gamine-2018   

          post_modified   post_modified_gmt post_parent  \
2   2019-12-21 09:00:17 2019-12-21 08:00:17         0.0   
4   2020-01-04 16:36:01 2020-01-04 15:36:01         0.0   
800 2019-12-21 09:00:17 2019-12-21 08:00:17         0.0   
802 2020-01-04 16:36:01 2020-01-04 15:36:01         0.0   

                                                  guid menu_order   post_type  \
2    https://www.bottle-neck.fr/?post_type=product&...        0.0     product   
4    https://www.bottle-neck.fr/?post_type=product&...        0.0     product   
800  https://www.bottle-neck.fr/wp-content/uploads/...        0.0  attachment   
802  https://www.bottle-neck.fr/wp-content/uploads/...        0.0  attachment   

    post_mime_type comment_count  
2              NaN           0.0  
4              NaN           0.0  
800     image/jpeg           0.0  
802     image/jpeg           0.0  

[4 rows x 21 columns]

La difference est que la premiere ligne correspond au produit(post_type=product) et la deuxieme corresponde à l'image (post_type=attachement). On supprime les lignes que correspondent à l'image. Et le deux colonne liées à elle(post_type et post_mime_type)

In [190]:
web.drop(web[web['post_type']=='attachment'].index, inplace=True)

In [191]:
web.nunique().tolist

<bound method IndexOpsMixin.tolist of sku                  714
average_rating         1
total_sales           41
tax_status             1
post_author            2
post_date            714
post_date_gmt        714
post_title           711
post_excerpt         677
post_status            1
comment_status         1
ping_status            1
post_name            714
post_modified        587
post_modified_gmt    587
post_parent            1
guid                 714
menu_order             1
post_type              1
post_mime_type         0
comment_count          1
dtype: int64>

On supprime aussi toutes les colonnes qui nous reste avec un valeur unique

In [192]:
web.drop(['average_rating','tax_status','post_status','comment_status','ping_status','post_parent','menu_order','post_type','post_mime_type','comment_count'],axis=1, inplace=True)

In [77]:
web.head()

sku  total_sales  post_author           post_date  \
0  bon-cadeau-25-euros         10.0          1.0 2018-06-01 13:53:46   
1                15298          6.0          2.0 2018-02-08 12:58:52   
2                15296          0.0          2.0 2018-02-08 13:49:41   
3                15300          0.0          2.0 2018-02-08 14:08:36   
4                19814          3.0          2.0 2018-02-09 14:01:05   

        post_date_gmt                                         post_title  \
0 2018-06-01 11:53:46                                  Bon cadeau de 25€   
1 2018-02-08 11:58:52        Pierre Jean Villa Saint-Joseph Préface 2018   
2 2018-02-08 12:49:41    Pierre Jean Villa Saint-Joseph Rouge Tildé 2017   
3 2018-02-08 13:08:36  Pierre Jean Villa Crozes-Hermitage Accroche Co...   
4 2018-02-09 13:01:05  Pierre Jean Villa IGP Collines Rhodaniennes Ga...   

                                        post_excerpt  \
0  <span style="color: #a85253;"><strong>Parlons ...   
1  Il exhale un nez subtil, racé avec du poivre f...   
2  Les vieilles vignes lui apportent une rare pro...   
3  Dentelle de fruit de jeunes syrah, aux tanins ...   
4  Gamine représente tout le fruité et la gourman...   

                                           post_name       post_modified  \
0                             bon-cadeau-de-25-euros 2018-06-01 14:13:57   
1        pierre-jean-villa-saint-joseph-preface-2018 2019-12-30 09:30:29   
2          pierre-jean-villa-saint-joseph-tilde-2017 2019-12-21 09:00:17   
3  pierre-jean-villa-croze-hermitage-accroche-coe... 2020-06-26 18:15:03   
4                  pierre-jean-villa-igp-gamine-2018 2020-01-04 16:36:01   

    post_modified_gmt                                               guid  
0 2018-06-01 12:13:57  https://www.bottle-neck.fr/?post_type=product&...  
1 2019-12-30 08:30:29  https://www.bottle-neck.fr/?post_type=product&...  
2 2019-12-21 08:00:17  https://www.bottle-neck.fr/?post_type=product&...  
3 2020-06-26 16:15:03  https://www.bottle-neck.fr/?post_type=product&...  
4 2020-01-04 15:36:01  https://www.bottle-neck.fr/?post_type=product&...

## erp

In [16]:
erp.head()

product_id  onsale_web  price  stock_quantity stock_status
0        3847           1   24.2               0   outofstock
1        3849           1   34.3               0   outofstock
2        3850           1   20.8               0   outofstock
3        4032           1   14.1               0   outofstock
4        4039           1   46.0               0   outofstock

In [17]:
erp.shape

(825, 5)

In [18]:
erp.isnull().sum()

product_id        0
onsale_web        0
price             0
stock_quantity    0
stock_status      0
dtype: int64

In [19]:
erp.nunique().tolist

<bound method IndexOpsMixin.tolist of product_id        825
onsale_web          2
price             381
stock_quantity    128
stock_status        2
dtype: int64>

In [315]:
erp.dtypes

product_id          int64
onsale_web          int64
price             float64
stock_quantity      int64
stock_status       object
dtype: object

In [318]:
round(erp.describe(),2)

product_id  onsale_web   price  stock_quantity
count      825.00      825.00  825.00          825.00
mean      5162.60        0.87   32.42           26.50
std        902.64        0.34   26.80           45.89
min       3847.00        0.00    5.20            0.00
25%       4348.00        1.00   14.60            1.00
50%       4907.00        1.00   24.40           11.00
75%       5805.00        1.00   42.00           34.00
max       7338.00        1.00  225.00          578.00

## liaison

In [319]:
liaison.head()

product_id id_web
0        3847  15298
1        3849  15296
2        3850  15300
3        4032  19814
4        4039  19815

In [193]:
liaison.rename(columns={'id_web': 'sku'}, inplace=True)

In [21]:
liaison.shape

(825, 2)

In [194]:
liaison.isnull().sum()

product_id     0
sku           91
dtype: int64

# Merge

In [196]:
lia_web=pd.merge(liaison,web, on='sku', how='inner')

In [118]:
lia_web.isnull().sum()

product_id           0
sku                  0
total_sales          0
post_author          0
post_date            0
post_date_gmt        0
post_title           0
post_excerpt         0
post_name            0
post_modified        0
post_modified_gmt    0
guid                 0
dtype: int64

In [197]:
lia_web.shape

(714, 12)

In [198]:
final=pd.merge(lia_web,erp, on='product_id', how='inner')

In [199]:
final.isnull().sum()

product_id           0
sku                  0
total_sales          0
post_author          0
post_date            0
post_date_gmt        0
post_title           0
post_excerpt         0
post_name            0
post_modified        0
post_modified_gmt    0
guid                 0
onsale_web           0
price                0
stock_quantity       0
stock_status         0
dtype: int64

In [200]:
final.shape

(714, 16)

In [37]:
final.head()

product_id    sku  total_sales  post_author           post_date  \
0        3847  15298          6.0          2.0 2018-02-08 12:58:52   
1        3849  15296          0.0          2.0 2018-02-08 13:49:41   
2        3850  15300          0.0          2.0 2018-02-08 14:08:36   
3        4032  19814          3.0          2.0 2018-02-09 14:01:05   
4        4039  19815          0.0          2.0 2018-02-12 09:04:37   

        post_date_gmt                                         post_title  \
0 2018-02-08 11:58:52        Pierre Jean Villa Saint-Joseph Préface 2018   
1 2018-02-08 12:49:41    Pierre Jean Villa Saint-Joseph Rouge Tildé 2017   
2 2018-02-08 13:08:36  Pierre Jean Villa Crozes-Hermitage Accroche Co...   
3 2018-02-09 13:01:05  Pierre Jean Villa IGP Collines Rhodaniennes Ga...   
4 2018-02-12 08:04:37          Pierre Jean Villa Côte Rôtie Carmina 2017   

                                        post_excerpt  \
0  Il exhale un nez subtil, racé avec du poivre f...   
1  Les vieilles vignes lui apportent une rare pro...   
2  Dentelle de fruit de jeunes syrah, aux tanins ...   
3  Gamine représente tout le fruité et la gourman...   
4  Le côte rôtie Carmina monte en puissance mais ...   

                                           post_name       post_modified  \
0        pierre-jean-villa-saint-joseph-preface-2018 2019-12-30 09:30:29   
1          pierre-jean-villa-saint-joseph-tilde-2017 2019-12-21 09:00:17   
2  pierre-jean-villa-croze-hermitage-accroche-coe... 2020-06-26 18:15:03   
3                  pierre-jean-villa-igp-gamine-2018 2020-01-04 16:36:01   
4          pierre-jean-villa-cote-rotie-carmina-2017 2020-01-04 16:36:10   

    post_modified_gmt                                               guid  \
0 2019-12-30 08:30:29  https://www.bottle-neck.fr/?post_type=product&...   
1 2019-12-21 08:00:17  https://www.bottle-neck.fr/?post_type=product&...   
2 2020-06-26 16:15:03  https://www.bottle-neck.fr/?post_type=product&...   
3 2020-01-04 15:36:01  https://www.bottle-neck.fr/?post_type=product&...   
4 2020-01-04 15:36:10  https://www.bottle-neck.fr/?post_type=product&...   

   onsale_web  price  stock_quantity stock_status  
0           1   24.2               0   outofstock  
1           1   34.3               0   outofstock  
2           1   20.8               0   outofstock  
3           1   14.1               0   outofstock  
4           1   46.0               0   outofstock

In [38]:
final.nunique().tolist

<bound method IndexOpsMixin.tolist of product_id           714
sku                  714
total_sales           41
post_author            2
post_date            714
post_date_gmt        714
post_title           711
post_excerpt         677
post_name            714
post_modified        587
post_modified_gmt    587
guid                 714
onsale_web             1
price                362
stock_quantity       126
stock_status           2
dtype: int64>

In [41]:
final[(final['stock_quantity']==0 )& (final['stock_status']=='instock')]

product_id                  sku  total_sales  post_author  \
410        4954  bon-cadeau-25-euros         10.0          1.0   

              post_date       post_date_gmt         post_title  \
410 2018-06-01 13:53:46 2018-06-01 11:53:46  Bon cadeau de 25€   

                                          post_excerpt  \
410  <span style="color: #a85253;"><strong>Parlons ...   

                  post_name       post_modified   post_modified_gmt  \
410  bon-cadeau-de-25-euros 2018-06-01 14:13:57 2018-06-01 12:13:57   

                                                  guid  onsale_web  price  \
410  https://www.bottle-neck.fr/?post_type=product&...           1   25.0   

     stock_quantity stock_status  
410               0      instock

In [123]:
final['CA']=(final['total_sales']*final['price'])

In [124]:
final.sort_values(by = 'CA', ascending = False).head()

product_id    sku  total_sales  post_author           post_date  \
194        4334   7818         96.0          2.0 2018-03-01 14:02:44   
71         4144   1662         87.0          2.0 2018-02-13 13:17:25   
218        4402   3510         13.0          2.0 2018-03-22 11:21:05   
70         4142  11641         30.0          2.0 2018-02-13 13:08:44   
69         4141    304         40.0          2.0 2018-02-13 12:57:44   

          post_date_gmt                              post_title  \
194 2018-03-01 13:02:44  Champagne Gosset Grand Blanc de Blancs   
71  2018-02-13 12:17:25             Champagne Gosset Grand Rosé   
218 2018-03-22 10:21:05                    Cognac Frapin VIP XO   
70  2018-02-13 12:08:44   Champagne Gosset Grand Millésime 2006   
69  2018-02-13 11:57:44         Champagne Gosset Grande Réserve   

                                          post_excerpt  \
194  La bulle fine et presente se dirige vers la su...   
71   Le nez exprime sans détour la fraise, dans sa ...   
218  La cuvée VIP XO à été enrichie d’eaux-de-vie t...   
70   L'attaque est ample, gourmande avec une belle ...   
69   Le nez, ouvert et expressif, évoque les fruits...   

                                 post_name       post_modified  \
194  champagne-gosset-grand-blanc-de-blanc 2020-08-12 14:00:03   
71             champagne-gosset-grand-rose 2020-08-14 16:45:03   
218                   cognac-frapin-vip-xo 2020-08-22 11:35:03   
70   champagne-gosset-grand-millesime-2006 2020-08-20 09:30:03   
69         gosset-champagne-grande-reserve 2020-08-27 11:25:02   

      post_modified_gmt                                               guid  \
194 2020-08-12 12:00:03  https://www.bottle-neck.fr/?post_type=product&...   
71  2020-08-14 14:45:03  https://www.bottle-neck.fr/?post_type=product&...   
218 2020-08-22 09:35:03  https://www.bottle-neck.fr/?post_type=product&...   
70  2020-08-20 07:30:03  https://www.bottle-neck.fr/?post_type=product&...   
69  2020-08-27 09:25:02  https://www.bottle-neck.fr/?post_type=product&...   

     onsale_web  price  stock_quantity stock_status      CA  
194           1   49.0               0   outofstock  4704.0  
71            1   49.0              11      instock  4263.0  
218           1  176.0               8      instock  2288.0  
70            1   53.0               8      instock  1590.0  
69            1   39.0               1      instock  1560.0

In [125]:
final['CA'].sum()

70568.6

In [177]:
final[final['total_sales']>5]

product_id    sku  total_sales  post_author           post_date  \
0          3847  15298          6.0          2.0 2018-02-08 12:58:52   
7          4042  16042          7.0          2.0 2018-02-12 09:54:27   
9          4045  16041         14.0          2.0 2018-02-12 10:09:03   
10         4046  15269          6.0          2.0 2018-02-12 10:16:27   
12         4048  16044          7.0          2.0 2018-02-12 10:34:08   
..          ...    ...          ...          ...                 ...   
651        6206  16580         41.0          2.0 2019-07-24 09:31:48   
652        6207  16077         37.0          2.0 2019-07-24 09:33:52   
660        6223  15324          6.0          2.0 2019-07-26 10:44:28   
678        6578  15647          7.0          2.0 2020-01-04 10:02:37   
684        6617  16031          6.0          2.0 2020-01-18 10:30:35   

          post_date_gmt                                         post_title  \
0   2018-02-08 11:58:52        Pierre Jean Villa Saint-Joseph Préface 2018   
7   2018-02-12 08:54:27                        Pierre Gaillard Cornas 2017   
9   2018-02-12 09:09:03                    Pierre Gaillard Côte Rôtie 2018   
10  2018-02-12 09:16:27       Pierre Gaillard Côte Rôtie Rose Pourpre 2017   
12  2018-02-12 09:34:08  Pierre Gaillard Saint-Joseph Rouge Clos de Cum...   
..                  ...                                                ...   
651 2019-07-24 07:31:48           Domaine Giudicelli Patrimonio Blanc 2019   
652 2019-07-24 07:33:52           Domaine Giudicelli Patrimonio Rouge 2016   
660 2019-07-26 08:44:28                    Planeta Sicilia Chardonnay 2017   
678 2020-01-04 09:02:37   Saumaize-Michelin Pouilly-Fuissé Ampélopsis 2016   
684 2020-01-18 09:30:35  Clos du Mont-Olivet Vins de Pays du Gard Rive ...   

                                          post_excerpt  \
0    Il exhale un nez subtil, racé avec du poivre f...   
7    Une jolie robe grenat avec des reflets violacé...   
9    Ce vin exprime la diversité et l'équilibre ent...   
10   La cuvée "Rose Pourpre" provient d'une parcell...   
12   D'une belle intensité, Clos de Cuminaille a de...   
..                                                 ...   
651  Ce Patrimonio Blanc signé Muriel Giudicelli no...   
652  100% Nielluccio ce Patrimonio présente un nez ...   
660  Une version 100% sicilienne du chardonnay. Un ...   
678  <div id="wrapper">\n<div id="container-wrapper...   
684  <span style="display: inline !important; float...   

                                            post_name       post_modified  \
0         pierre-jean-villa-saint-joseph-preface-2018 2019-12-30 09:30:29   
7                         pierre-gaillard-cornas-2017 2020-08-14 10:15:02   
9                     pierre-gaillard-cote-rotie-2018 2020-08-03 09:55:03   
10       pierre-gaillard-cote-rotie-rose-pourpre-2017 2020-08-24 14:00:03   
12       pierre-gaillard-saint-joseph-cuminaille-2018 2020-08-14 17:45:03   
..                                                ...                 ...   
651          domaine-giudicelli-patrimonio-blanc-2019 2020-08-27 10:12:58   
652          domaine-giudicelli-patrimonio-rouge-2016 2020-08-25 11:55:02   
660                   planeta-sicilia-chardonnay-2017 2020-07-30 11:35:02   
678  saumaize-michelin-pouilly-fuisse-ampelopsis-2016 2020-05-24 09:17:20   
684     clos-du-mont-olivet-igp-gard-rive-droite-2018 2020-08-07 10:15:02   

      post_modified_gmt                                               guid  \
0   2019-12-30 08:30:29  https://www.bottle-neck.fr/?post_type=product&...   
7   2020-08-14 08:15:02  https://www.bottle-neck.fr/?post_type=product&...   
9   2020-08-03 07:55:03  https://www.bottle-neck.fr/?post_type=product&...   
10  2020-08-24 12:00:03  https://www.bottle-neck.fr/?post_type=product&...   
12  2020-08-14 15:45:03  https://www.bottle-neck.fr/?post_type=product&...   
..                  ...                                                ...   
651 2020-08-27 08:12:58  https://ww

In [132]:
round(final['price'].describe(),2)

count    714.00
mean      32.49
std       27.81
min        5.20
25%       14.10
50%       23.55
75%       42.18
max      225.00
Name: price, dtype: float64